In [28]:
import pandas as pd
import numpy as np

In [8]:
data = pd.read_csv("nutrition_values.csv", sep=';')
data.head()

Chain                                 Item                Type  \
0  Burger King                     Whopper Sandwich  Whopper Sandwiches   
1  Burger King         Whopper Sandwich with Cheese  Whopper Sandwiches   
2  Burger King      Bacon & Cheese Whopper Sandwich  Whopper Sandwiches   
3  Burger King              Double Whopper Sandwich  Whopper Sandwiches   
4  Burger King  Double Whopper Sandwich with Cheese  Whopper Sandwiches   

  Serving Size (g)  Calories  Calories from fat Total Fat (g)  \
0              270       660                360            40   
1              292       740                420            46   
2              303       790                460            51   
3              354       900                520            58   
4              377       980                580            64   

  Saturated Fat (g) Trans Fat (g)  Chol (mg)  Sodium (mg)  Total Carb (g)  \
0                12           1,5         90          980              49   
1                16             2        115         1340              50   
2                17             2        125         1560              50   
3                20             3        175         1050              49   
4                24             3        195         1410              50   

   Dietary Fiber (g)  Total Sugar (g)  Protein (g)  
0                  2               11           28  
1                  2               11           32  
2                  2               11           35  
3                  2               11           48  
4                  2               11           52

In [9]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 499 entries, 0 to 498
Data columns (total 15 columns):
 #   Column             Non-Null Count  Dtype 
---  ------             --------------  ----- 
 0   Chain              499 non-null    object
 1   Item               499 non-null    object
 2   Type               499 non-null    object
 3   Serving Size (g)   241 non-null    object
 4   Calories           499 non-null    int64 
 5   Calories from fat  499 non-null    int64 
 6   Total Fat (g)      499 non-null    object
 7   Saturated Fat (g)  499 non-null    object
 8   Trans Fat (g)      499 non-null    object
 9   Chol (mg)          499 non-null    int64 
 10  Sodium (mg)        499 non-null    int64 
 11  Total Carb (g)     499 non-null    int64 
 12  Dietary Fiber (g)  499 non-null    int64 
 13  Total Sugar (g)    499 non-null    int64 
 14  Protein (g)        499 non-null    int64 
dtypes: int64(8), object(7)
memory usage: 58.6+ KB


First observation:
- There are 7 object type variables - ['Chain', 'Item', 'Type', 'Serving Size (g)', 'Total Fat (g)', 'Saturated Fat (g)', 'Trans Fat (g)', and the rest are all numerical

## Data Preprocess

In [12]:
#check null values
data.isna().sum()

Chain                  0
Item                   0
Type                   0
Serving Size (g)     258
Calories               0
Calories from fat      0
Total Fat (g)          0
Saturated Fat (g)      0
Trans Fat (g)          0
Chol (mg)              0
Sodium (mg)            0
Total Carb (g)         0
Dietary Fiber (g)      0
Total Sugar (g)        0
Protein (g)            0
dtype: int64

There are 258 NA rows in 'Serving Size' column, let's take a look at the information stored in this column 'Serving Size (g)' 

In [18]:
data[['Chain','Item','Serving Size (g)']].head(20)

Chain                                 Item Serving Size (g)
0   Burger King                     Whopper Sandwich              270
1   Burger King         Whopper Sandwich with Cheese              292
2   Burger King      Bacon & Cheese Whopper Sandwich              303
3   Burger King              Double Whopper Sandwich              354
4   Burger King  Double Whopper Sandwich with Cheese              377
5   Burger King              Triple Whopper Sandwich              438
6   Burger King  Triple Whopper Sandwich with Cheese              461
7   Burger King                  Whopper JR Sandwich              134
8   Burger King                  Bacon King Sandwich              356
9   Burger King          Cheddar Bacon King Sandwich              366
10  Burger King   Single Quarter Pound King Sandwich              231
11  Burger King   Double Quarter Pound King Sandwich              338
12  Burger King                  Single Stacker King              223
13  Burger King                  Double Stacker King              334
14  Burger King          Impossible Whopper Sandwich              285
15  Burger King                            Hamburger               99
16  Burger King                         Cheeseburger              111
17  Burger King                     Double Hamburger              136
18  Burger King                  Double Cheeseburger              148
19  Burger King              Extra Long Cheeseburger              226

Since the intention of this analysis is on sugar and calories analysis, the serving size, trans fat and saturated fat are that necessary, we would then exclude these columns out of the dataframe.

In [19]:
new_data = data.drop(['Serving Size (g)','Calories from fat','Saturated Fat (g)','Trans Fat (g)'], axis=1)
new_data.head()

Chain                                 Item                Type  \
0  Burger King                     Whopper Sandwich  Whopper Sandwiches   
1  Burger King         Whopper Sandwich with Cheese  Whopper Sandwiches   
2  Burger King      Bacon & Cheese Whopper Sandwich  Whopper Sandwiches   
3  Burger King              Double Whopper Sandwich  Whopper Sandwiches   
4  Burger King  Double Whopper Sandwich with Cheese  Whopper Sandwiches   

   Calories Total Fat (g)  Chol (mg)  Sodium (mg)  Total Carb (g)  \
0       660            40         90          980              49   
1       740            46        115         1340              50   
2       790            51        125         1560              50   
3       900            58        175         1050              49   
4       980            64        195         1410              50   

   Dietary Fiber (g)  Total Sugar (g)  Protein (g)  
0                  2               11           28  
1                  2               11           32  
2                  2               11           35  
3                  2               11           48  
4                  2               11           52

In [20]:
new_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 499 entries, 0 to 498
Data columns (total 11 columns):
 #   Column             Non-Null Count  Dtype 
---  ------             --------------  ----- 
 0   Chain              499 non-null    object
 1   Item               499 non-null    object
 2   Type               499 non-null    object
 3   Calories           499 non-null    int64 
 4   Total Fat (g)      499 non-null    object
 5   Chol (mg)          499 non-null    int64 
 6   Sodium (mg)        499 non-null    int64 
 7   Total Carb (g)     499 non-null    int64 
 8   Dietary Fiber (g)  499 non-null    int64 
 9   Total Sugar (g)    499 non-null    int64 
 10  Protein (g)        499 non-null    int64 
dtypes: int64(7), object(4)
memory usage: 43.0+ KB


In [36]:
new_data['Total Sugar (g)'].unique()

array([ 11,   7,  10,  12,   6,   9,   8,   5,  13,   0,   1,   4,   2,
         3,  25,  20,  24,  22,  30,  43,  33,  15,  40,  31,  18,  50,
        98, 101,  85,  88,  99,  58,  73, 105, 138,  56,  70, 102, 133,
        51,  64,  93, 121,  65,  81, 118, 155,  61,  76, 110, 145,  77,
       111, 146,  60,  75, 109, 143,  35,  49,  71,  21,  27,  34,  32,
        41,  16,  17,  14,  45,  44,  48,  97,  63,  84, 168,  59,  79,
       100, 124,  82, 103,  54,  72,  96,  39,  29,  86,  28,  83,  94,
        46,  69,  36,  53,  37,  55,  42,  47,  26,  38,  57,  87],
      dtype=int64)

In [24]:
new_data["Total Fat (g)"].unique()

array(['40', '46', '51', '58', '64', '75', '82', '18', '79', '84', '29',
       '54', '42', '68', '34', '10', '13', '21', '37', '16', '24', '41',
       '52', '49', '30', '11', '27', '15', '22', '74', '17', '9', '28',
       '25', '33', '4', '8', '2,5', '3', '0', '4,5', '14', '19', '20',
       '5', '38', '45', '26', '23', '44', '31', '3,5', '6', '1', '12',
       '39', '2', '7', '1,5', '32', '48', '56', '60', '36', '0,5'],
      dtype=object)

In [34]:
new_data[["Total Fat (g)"]].astype(str)

Total Fat (g)
0              40
1              46
2              51
3              58
4              64
..            ...
494             1
495           0,5
496             1
497             1
498           0,5

[499 rows x 1 columns]

In [35]:
new_data["Total Fat (g)"].str.replace(",", ".")

0       40
1       46
2       51
3       58
4       64
      ... 
494      1
495    0.5
496      1
497      1
498    0.5
Name: Total Fat (g), Length: 499, dtype: object

In [42]:
new_data.to_csv(r"C:\Users\Big City Boy\Desktop\Tableau input\new_nutrition_values.csv")
new_data.to_csv("new_nutrition_values.csv")